modules/images/webp-uploads/load.php

```diff
@@ -90,7 +90,13 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 
 		$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
 		$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
-		$image       = webp_uploads_generate_additional_image_source( $attachment_id, 'full', $original_size_data, $targeted_mime, $destination );
+
+		// Skip creation of duplicate WebP image if a image file has the same name with an extension of jpe, jpg, or jpeg.
+		if ( file_exists( $destination ) ) {
+			continue;
+		}
```

Comment:
Instead of adding this check here, I think a cleaner solution would be to handle it within the `webp_uploads_generate_additional_image_source()` function which is the function where the file is actually generated. That would fix the problem at its root and it would ensure you don't have to add a condition around it everywhere you call that function.

The current implementation here is actually missing the scenario when a new image size is generated where that function is also called - right now there would still be a chance for the image to be overwritten. By adding the condition to `webp_uploads_generate_additional_image_source()` and returning a `WP_Error` in that case, we only need to make that change there and it will work everywhere without any further changes.